### DISCUSSION TITLE
Interim search filter redesign

### DISCUSSION THREAD

**Discussion by rhystmills** [guardian/grid#3934] [Ideas]
TODO list:
- [ ] https://github.com/guardian/grid/issues/3975
- [ ] https://github.com/guardian/grid/pull/3974

The BBC's Grid contributors (especially UX) have been working on some interim redesigns for the Grid's filter system (currently based on 'chips').

(Plans for the future reimagination of search can be discussed in #3957 )

As we are planning to move on from AngularJS in the long run, it would make sense to do new work in React via `react2angular`.

It would be nice to manage the 'version' of search filters via a feature switch, to enable user testing in production Grid.

We'd also like to add telemetry, so that we can track how users are using filters (both now and after the redesign) - which ones are used commonly, and which aren't used very much at all.

To me, the sensible order of work is:
1. Move existing search component to React
2. Add a feature switch panel
3. Add telemetry
4. Add the new design behind a feature switch (non dev work can take place in parallel with above work)

Does this sound sensible to others?

**Comment by RichardLe** [comment]
Discussed with on 16/11
* BBC To add URLS for searches on their filter list (Usable By All, For News, Chargeable, For Programme Promotions, My Uploads)
* BBC will also add more crafted designs for the UX
<img width="982" alt="Screenshot 2022-11-16 at 12 58 18" src="https://user-images.githubusercontent.com/1539653/202187257-a92cca32-3363-4fe9-8d1a-7f4960e0cc60.png">

**Comment by paperboyo** [comment]
From Miro:
- Usable for all: `search?query=is:BBC-owned-photo -has:restrictions` (maybe you would want to add `-leases.leases.access:deny-use`, which is not perfect, because https://github.com/guardian/grid/issues/3860, and because adding `leases.leases.active:true` to it sadly will catch **any** active lease, not just `deny` one(s), but still?)
- For News: [to come]
- Chargeable: `search?nonFree=true` (yeah, no way of easily searching for all chageable/restricted/denied/No rights)
- For promotion programme: [to come]

Some points from/around the discussion about interim solution(s) before I forget:

1.
- MK: hiding all filters behind dropdown:
    - gets rid of very important Free to use (for both editorial and rights) from immediate view (scary)
    - Adds additional click to (un)check it (and this other one not many people use and no one should much care about extra click)
    - how ’bout following current “responsive” pattern of showing at least Free to use if there is space and hiding **the rest** behind the dropdown?
- LG: yeah, but that’s inconsistent (MK: yep, will look different depending on width)
2. MK: My uploads is a filter, My recent uploads is a link to Upload page. Forgetting for now that Upload page should die one day™ (search page should have a density adjustment and be able to show metadata like current Upload page does), this is confusing in naming. And makes it difficult to figure out which one should go under avatar user menu (once we build it). My idea would be something like: My uploads goes under avatar and My recent uploads can… be renamed (?) and go… also under avatar? outside? Anyhow: worth discussing
3. Avatar as a place for storing “my stuff” got general applause. See also https://github.com/guardian/grid/issues/539#issuecomment-404420763. One day™ we should review where/when we do store a user and a date for action and where we need to add it/them. For Last modified by me etc… My cropped we can add now (but not sort by date cropped yet, even though we record it, phew)
4. MK: Sorting should get separate ordering device (value agnostic) and a value dropdown. But makes little sense to touch before tackling sorting by Date taken. In order to do that, we need to decide what to do with partial and with missing dates (separately: we should allow entering partials as we are now forcing users to enter wrong data for partials). So this feels like not in Interim?
5. Important for Gu is GNM owned and (maybe) Hide over-quota. Users make a very good point that both of them should be contextual. And that GNM owned should be in your face. So much so that it must be visible at all times. Their argument goes: _we have great photography we paid lotsa for, sent snapper for an event, but [enter non-picdesk-users] always seem to find some borring agency picture_. This argument sounds to us very strong. Hide over-quota is much less strong and possibly can be behind the Filter dropdown (problem then: won’t be contextual, so if quotas aren’t over, will do nothing; possible solution: grey it out in Filters dropdown if not applicable? also won’t get a count). I have proposed reusing the New ticker for GNM owned. LG said it doesn’t look like a button (agreed, but applies to current ticker too). But I think we would still like a solution that satisfies the above points about GNM owned and being one of many filters behind dropdown doesn’t do this. Hence:
    - could you think if you would ever want any filter(s) to have properties like described above for GNM owned (prominent, dynamic, showing counts)?
    - even if not, could you think of a better way of achieving this without reusing the ticker? (I struggle; and reusing ticker isn’t 100% agreeable to me either) Help appreciated!
6. Not sure if I mentioned this, but filters important to you from Miro sound like they try to do multi tenancy/integration on the cheap. I understand the immediate value for users! But I feel we are introducing more and more plasters, that will have to be unlearned and removed (at least some) after proper multi tenancy is done one day™. So just wanted to say ceterum autem censeo integration, integration, integration ;-)

**Comment by paperboyo** [comment]
Ad. 6 From Grid Hour 21/11: @louisegoodspeed  and @honorcb pointed out very important modification to [org]-owned: the UI device should probably not surface restricted and denied [org]-owned pics?

- Another potential candidate for this type of prominent UI device: agency/user picks (once we have them and give them dynamic borders etc)

**Comment by louisegoodspeed** [comment]
New ideas around designs for this can be found on the Miro board link: https://miro.com/app/board/uXjVPIgBRFQ=/ Password: [message one of us to get this]

Designs need to be perfected - @louisegoodspeed 
 - sort by - show ascending/decending
 - look at how we can merge profile with the dots/setttings to save space and mock up what is under profile..
 - Check spacing, clickable areas etc
 
 Decisions still needed: @paperboyo @andrew-nowak @CDicksonG @rhystmills @RichardLe 
 - Do the Guardian want to go with the tick boxes as well as the ticker link for GNM owned? Or just ticker?
 - Are we going to partner up on the development?
 - What will be in React vs Angular JS?
  
@louisegoodspeed  and @honorcb  to come back for the searches for permissions filters

**Comment by louisegoodspeed** [comment]
Notes from meeting 25th January 
 - @RichardLe talked about Multi-tenancy and the BBC direction -not hiding things from users and based on what a user is trying to do. on We're researching and thinking things through with user cases in mind  @andrew-nowak is keen to be involved in these conversations
- @louisegoodspeed shared updated designs on the Miro board
- Agreed we'd go ahead with the designs proposed. There was concern about the Taken Date and Sort by, so @paperboyo is going to set up a separate discussion about this.
- The Guardian want to go for a tickbox rather than the ticket button for GNM owned
- We agreed to write down a discussion on the requirements for the designs - it'll probably be a combined effort but @louisegoodspeed will kick this off
- We agreed we'd want to do build in Angular to React
- Then we need to work out the Who/When
  
  Please add anything else I've missed.

**Comment by louisegoodspeed** [comment]
Update from BBC due to the filters meeting being cancelled today.

- UX have been carrying out more Interviews and found more instances of double hats. For instance the same user could be working on updating the BBC Homepage with a news story and then working on BBC Sounds. So this confirms we need a multi-tenancy solution that works for the user case rather than the user.
- We tested our first prototype of the interim filter designs with one user. 
		- They understood what each of the drop down items meant under Permissions . They felt they were clear and what was 
		   needed so they would be able to find the images they could choose from 	
                   depending on their hat that day.
		 - They found the sort by and new how to change that
		 - Also found their account area
“this seems less cluttered and easier to use”
We’ll do some more testing as we shadow/interview people.
 	
 - We’ve had a few discussions re the sort by but need to respond to the GitHub discussion - broadly in agreement and great Ana will investigate points 5-11. @pradasa maybe we can catch up on this in our monthly chat.
 
 - I’ve gone through the filters survey and pulled out findings. Next steps are to present to BBC Images team and then we can share that with you 
 
 - I started a paper doc on the interim filters design, we’ll put that into GitHub discussion when BBC folks are agreed all the right info is in there to enable the build to start.

@louisegoodspeed is on leave on Monday, so won’t be at GridHour.
@RichardLe @davmacbea just tagging you so you can amend, change anything if needed...
